File: tests/sdk/nodejs/test/ssm.test.mjs
import { test } from "node:test";
import assert from "node:assert/strict";
import {
  DeleteParameterCommand,
  GetParameterCommand,
  PutParameterCommand,
} from "@aws-sdk/client-ssm";
import { newSSM, requireReady, uniquePrefix } from "../lib/helpers.mjs";

test("SSM String parameter round-trip", async (t) => {
  if (!(await requireReady(t))) return;
  const ssm = newSSM();
  const prefix = uniquePrefix();
  const name = `/lab/${prefix}/config`;
  const value = `plain-${prefix}`;

  await ssm.send(
    new PutParameterCommand({
      Name: name,
      Type: "String",
      Value: value,
    }),
  );
  t.after(async () => {
    try {
      await ssm.send(new DeleteParameterCommand({ Name: name }));
    } catch {
      /* ignore */
    }
  });

  const got = await ssm.send(new GetParameterCommand({ Name: name }));
  assert.equal(got.Parameter?.Value, value);

  await ssm.send(new DeleteParameterCommand({ Name: name }));
});

File: tests/sdk/go/ssm_test.go
package sdk_test

import (
	"context"
	"testing"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/service/ssm"
	"github.com/aws/aws-sdk-go-v2/service/ssm/types"
)

func TestSSMStringParameterRoundTrip(t *testing.T) {
	requireReady(t)
	cfg := loadAWSConfig(t)
	client := newSSM(t, cfg)
	ctx := context.Background()
	prefix := uniquePrefix(t)
	name := "/lab/" + prefix + "/config"
	value := "plain-" + prefix

	_, err := client.PutParameter(ctx, &ssm.PutParameterInput{
		Name:  aws.String(name),
		Type:  types.ParameterTypeString,
		Value: aws.String(value),
	})
	if err != nil {
		t.Fatalf("PutParameter: %v", err)
	}
	t.Cleanup(func() {
		_, _ = client.DeleteParameter(ctx, &ssm.DeleteParameterInput{Name: aws.String(name)})
	})

	got, err := client.GetParameter(ctx, &ssm.GetParameterInput{Name: aws.String(name)})
	if err != nil {
		t.Fatalf("GetParameter: %v", err)
	}
	if got.Parameter == nil || aws.ToString(got.Parameter.Value) != value {
		t.Fatalf("GetParameter unexpected: %+v", got.Parameter)
	}

	_, err = client.DeleteParameter(ctx, &ssm.DeleteParameterInput{Name: aws.String(name)})
	if err != nil {
		t.Fatalf("DeleteParameter: %v", err)
	}
}
